### ISSUE TITLE
SMHasher results as a quality metric are a bit flat, try SMHasher3, maybe?

### ISSUE THREAD

**Issue by sh1boot** [Cyan4973/xxHash#890]
[SMHasher3](https://gitlab.com/fwojcik/smhasher3/) seems to test nearer "to destruction" than the earlier forks, so there's something more interesting than just a bunch of tens to for comparisons with other hashes:
https://gitlab.com/fwojcik/smhasher3/-/blob/main/results/XXH-64.txt#L3032
```
Overall result: FAIL            ( 237 / 246 passed)
Failures:
    SeedBlockLen        : [15, 19, 21, 26, 29, 30]
    SeedBlockOffset     : [8]
    SeedBIC             : [4, 8]
```

It's GPLv3, though, so not universally accessible.

**Comment by Cyan4973** [comment]
I recently tried to use it, but ended up on a compilation error :
```
In file included from smhasher3/util/Random.cpp:19:
smhasher3/util/Random.h: In member function 'void Rand::update_xseed()':
smhasher3/build/include/Platform.h:83:23: error: '_rotr64' was not declared in this scope
   83 | #define ROTR64(v, n) (_rotr64((v), (n)))
      |                       ^~~~~~~
smhasher3/util/Random.h:221:22: note: in expansion of macro 'ROTR64'
  221 |         xseed[3] = ((ROTR64(rseed, 32) | 1) * M2);
      |                      ^~~~~~
```

**Comment by sh1boot** [comment]
Try using GCC.  I don't think the author uses clang for testing and incompatibilities creep back in from time to time.

**Comment by Cyan4973** [comment]
I could run SMHasher3 on a different workstation, where it finally worked fine.

After testing it a bit, I think it's a good fork, well maintained, and better optimized. 
It does employ the improved performance to extend analysis to much finer grain details, requiring random properties not just at hash level but across all sub-ranges, and also pushing bit independence tests into more scenarios (they used to be extremely slow in the original version of SMHasher, making them less useful).

Under this new set of constraints, non-cryptographic hashes have a hard time passing all the tests. 
One could even wonder if a hash function need to be so perfect. Maybe there's a point after which the additional randomness property is less important than, for example, speed. But that's a debate for another post.

I'll keep an eye on this fork in the future, and may switch to using it at some point.

**Comment by sh1boot** [comment]
Well I can't speak to the importance of such challenging tests.  I just felt that it was more interesting than a competition where everybody scored 100%.

But if more people were aware of version 3, would every hash soon ace it, leaving us with the same situation all over again?

Certainly I tuned my own hash until it got 100%, then optimised it until it broke, then did a survey of magic numbers until I found one that scored 100% again.

"There, I fixed it!"
